doc/source/dev/conduct/code_of_conduct.rst

```diff
@@ -0,0 +1,164 @@
+
+
+Diversity Statement
+-------------------
+
+The SciPy project welcomes and encourages participation by everyone. We are
+committed to being a community that everyone feels good about joining. Although
+we may not be able to satisfy everyone, we will always work to treat everyone
+well.
+
+No matter how you identify yourself or how others perceive you: we welcome you.
+Though no list can hope to be comprehensive, we explicitly honour diversity in:
+age, culture, ethnicity, genotype, gender identity or expression, language,
+national origin, neurotype, phenotype, political beliefs, profession, race,
+religion, sexual orientation, socioeconomic status, subculture and technical
+ability.
+
+Though we welcome people fluent in all languages, SciPy development is
+conducted in English.
+
+Standards for behaviour in the SciPy community are detailed in the Code of
+Conduct above. We expect participants in our community to meet these standards
+in all their interactions and to help others to do so as well.
+
+
+Reporting Guidelines
+--------------------
+
+While this code of conduct should be adhered to by participants, we recognize
+that sometimes people may have a bad day, or be unaware of some of the
+guidelines in this code of conduct. When that happens, you may reply to them
+and point out this code of conduct. Such messages may be in public or in
+private, whatever is most appropriate. However, regardless of whether the
+message is public or not, it should still adhere to the relevant parts of this
+code of conduct; in particular, it should not be abusive or disrespectful.
+Assume good faith; it is more likely that participants are unaware of their bad
+behaviour than that they intentionally try to degrade the quality of the
+discussion.
+
+Should there be difficulties in dealing with the situation, you may report your
+compliance issues in confidence to either:
+
+- The SciPy Code of Conduct Committee, at conduct@TODO (preferred)
+  Members of this committee are TODO
+
+Or alternatively to:
+
+- Chair of the SciPy Steering Committee: Ralf Gommers
+- Executive Director of NumFOCUS: Leah Silen
+
+
+Incident reporting resolution & Code of Conduct enforcement
+-----------------------------------------------------------
+
+*This section summarizes the most important points, more details can be found
+in* :ref:`CoC_reporting_manual`.
+
+All complaints will be reviewed and investigated and will result in a response
+that is deemed necessary and appropriate to the circumstances. The SciPy Code
+of Conduct Committee, and the SciPy Steering Committee (if involved), are
+obligated to maintain confidentiality with regard to the reporter of an
+incident. 
+
+The process for acting on any received code of conduct violation report at a
+high level (for all but the most severe cases) will be:
+
+1. acknowledge report is received
+2. reasonable discussion/feedback
+3. mediation (if feedback didn't help)
```

Comment:
I'm very concerned about this "mediation" thing.

Mediation can be a very valuable thing. But I don't see how it has anything to do with CoC violations.

Mediation is useful in helping two or more folks resolve an intractable disagreement. Normally (I hope!) such disagreements don't involve any CoC violations. So a weird thing is that right now, it seems like this is a generally useful thing that we only reward people with if they behave badly? OTOH, if there is a CoC violation, then that needs to be addressed separately. E.g. if some maintainer starts screaming at another in public, then it's not just a matter for the two of them to work out -- it also affects the whole tone of the project, intimidates bystanders from contributing, etc., and we need to respond by making clear that this is unacceptable. It's great if the two people most directly involved are able to work things out, but that doesn't really solve the problem.

And then, similar to my other big comment above, the text here seems to be saying that actual bad faith abuse/harassment simply doesn't happen, and that we won't handle it appropriately if it does. It's an actual cliche that in cases of sexist harassment, a common harmful response is for authorities to insist that the offender and victim work things out, as if the victim could somehow have stopped the rape threats by having a sensible conversation, or that they need to come to some accomodation for how someone else is claiming credit for their work. Plus it puts the onus on the victim to somehow fix the offender, which is hardly fair.

The report handling manual is less terrible -- at least it has basic safeguards like emphasizing that this is a purely voluntary process. But the text here -- in the document for folks who've just been victimized -- doesn't have any of that; instead it gives the impression that this is the only approach we'll consider (except in "the most severe cases", whatever that means – I guess what it will mean to many readers is that we'll be judging each victim's report to decide whether they've been harmed "enough" to warrant a more appropriate response). And even with the safeguards, I'm still extremely uncomfortable about even asking a victim if they're willing to participate in this kind of mediation, because no matter how we do it there's a risk that we'll be putting pressure on them to participate in something that they're not really comfortable with.

And in any case, like I said above, I think we'd be better off dropping the whole mediation discussion entirely, or moving it into some document that's separate from the CoC documents.

Reply:
I imagine the author and you hold different examples in their heads while writing this.  An expletive, say, vs sexual harassment.  As in my previous comment, I suggest that the tool used to resolve the conflict corresponds to the offense.  E.g., swearing a lot may need some cool off time from the mailing list, and apologizing to someone sworn at.  In the other example, it is likely there is no good way for the offender to remain part of the community.

Mediation is for a certain class of less severe problems: misunderstandings, not giving due credit, project management frustrations, etc.  Basically, those that can be handled by sitting down around the table and talking it out with some guidance.

I think it's fair to expect the language to be very clear w.r.t. whether mediation is applicable, and asserting that it is not required.  For the committee, a guideline around how to approach mediation may also be helpful.

Reply:
Please have a look at the current version.